### PR TITLE
Add UCSBDiningCommonsMenuItem Table, Fixtures and Relevant Tests

### LINKED ISSUE #25: Add MenuItemTable component to Storybook (along with fixtures)
- [x] There is a folder under frontend/src/stories/components with the name MenuItem with a file inside it called MenuItemTable.stories.js
- [x] There is a file under the frontend/src/fixtures folder with the name menuItemFixtures.js
- [x] There are two fixtures inside the menuItemFixtures.js file: one with one MenuItem and another one with three MenuItems
- [x] When the fixtures are viewed on Storybook, they appear as expected (empty one should not have anything, the one with three MenuItems should have all fields set as expected).

### LINKED ISSUE #26: Replace MenuItemTable placeholder page with one that includes MenuItemTable
Replace RecommendationTable placeholder page with one that includes RecommendationTable.

- [x] Replace placeholder code with code from the Javascript index page
- [x] Check if the index page is correctly connected to the correct backend API
- [x] Check if the actual table shows up on the front end
- [x] There is full test coverage (Jacoco) for RecommendationRequestController
- [x] There is full mutation test coverage (Pitest) for the newly added code

### LINKED ISSUE #27: Add Delete Column to MenuItemTable (if not already done)

### PR DESCRIPTION
**Pull request by ryang1337**
Created table for UCSBDiningCommonsMenuItem and replaced placeholder index page so that it displays the table. Also added delete functionality for Admin users.

Created fixtures for UCSBDiningCommonsMenuItem and added stories to display fixtures on Storybook

Created tests to check if components renders and to make sure delete functionality works.

100% coverage with `npm run coverage`
mutation score of 100 with `npx stryker run`

Closes #25 #26 #27

### CODE REVIEW

File: frontend/src/main/utils/UCSBDiningCommonsMenuItemUtils.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { toast } from "react-toastify";
+
+export function onDeleteSuccess(message) {
+    console.log(message);
+    toast(message);
+}
+
+export function cellToAxiosParamsDelete(cell) {
```
Comment (yuheho7749) [inline frontend/src/main/utils/UCSBDiningCommonsMenuItemUtils.js:8]: If you have the cellToAxiosParamsDelete here, you need to import and use it in the UCSBDiningCommonsMenuItemTable. Also, the api url is supposed to be the camel cased one

File: frontend/src/main/components/UCSBDiningCommonsMenuItem/UCSBDiningCommonsMenuItemTable.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import OurTable, { ButtonColumn } from "main/components/OurTable";
+import { useBackendMutation } from "main/utils/useBackend";
+import { onDeleteSuccess } from "main/utils/UCSBDiningCommonsMenuItemUtils"
+// import { useNavigate } from "react-router-dom";
+import { hasRole } from "main/utils/currentUser";
+
+
+export function cellToAxiosParamsDelete(cell) {
```
Comment (yuheho7749) [inline frontend/src/main/components/UCSBDiningCommonsMenuItem/UCSBDiningCommonsMenuItemTable.js]: Duplicate function definition in UCSBDiningCommonsMenuItemUtils

File: frontend/src/stories/components/UCSBDiningCommonsMenuItem/UCSBDiningCommonsMenuItem.stories.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
```
Comment (yuheho7749) [inline frontend/src/stories/components/UCSBDiningCommonsMenuItem/UCSBDiningCommonsMenuItem.stories.js]: Not sure if the file name "UCSBDiningCommonsMenuItem" should be "UCSBDiningCommonsMenuItemTable"

File: frontend/src/tests/utils/UCSBDiningCommonsMenuItemUtils.test.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        test("It returns the correct params", () => {
+            // arrange
+            const cell = { row: { values: { id: 17 } } };
+
+            // act
+            const result = cellToAxiosParamsDelete(cell);
+
+            // assert
+            expect(result).toEqual({
+                url: "/api/ucsbdiningcommonsmenuitem",
```
Comment (yuheho7749) [inline frontend/src/tests/utils/UCSBDiningCommonsMenuItemUtils.test.js]: api url should be camel cased?

**Review by yuheho7749 — CHANGES_REQUESTED** [review]
A few minor things needs to be changed (naming/typos)

File: frontend/src/main/utils/UCSBDiningCommonsMenuItemUtils.js (hunk 1 above)
Comment (ryang1337) [reply frontend/src/main/utils/UCSBDiningCommonsMenuItemUtils.js:8]: fixed

File: frontend/src/main/components/UCSBDiningCommonsMenuItem/UCSBDiningCommonsMenuItemTable.js (hunk 2 above)
Comment (ryang1337) [reply frontend/src/main/components/UCSBDiningCommonsMenuItem/UCSBDiningCommonsMenuItemTable.js]: fixed

File: frontend/src/stories/components/UCSBDiningCommonsMenuItem/UCSBDiningCommonsMenuItem.stories.js (hunk 3 above)
Comment (ryang1337) [reply frontend/src/stories/components/UCSBDiningCommonsMenuItem/UCSBDiningCommonsMenuItem.stories.js]: fixed

File: frontend/src/tests/utils/UCSBDiningCommonsMenuItemUtils.test.js (hunk 4 above)
Comment (ryang1337) [reply frontend/src/tests/utils/UCSBDiningCommonsMenuItemUtils.test.js]: fixed

**Review by yuheho7749 — APPROVED** [review]
LGTM

**Pull request merged**